src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -145,8 +145,22 @@ public async Task RemoveAsync(IAccount account)
 
         private async Task<IEnumerable<IAccount>> GetAccountsInternalAsync(ApiIds apiId, string homeAccountIdFilter = null)
         {
-            var accountsFromCache = await GetAccountsFromCacheAsync(apiId, homeAccountIdFilter).ConfigureAwait(false);
-            var accountsFromBroker = await GetAccountsFromBrokerAsync(homeAccountIdFilter).ConfigureAwait(false);
+            RequestContext requestContext = CreateRequestContext(Guid.NewGuid());
+
+            var authParameters = new AuthenticationRequestParameters(
+                   ServiceBundle,
```

Comment:
could you just call `CreateRequestParameters(new AcquireTokenCommonParameters() { ApiId = apiId }, CreateRequestContext(Guid.NewGuid()), UserTokenCacheInternal);` and pass it directly to the CacheSessionManager?